dora/core/server/worker/src/main/java/alluxio/worker/dora/DoraMetaManager.java

```diff
@@ -0,0 +1,159 @@
+   * @return the file status, or empty optional if not found
+   */
+  public Optional<FileStatus> getFromUfs(String path) throws IOException {
+    try {
+      UfsStatus status = mUfs.getStatus(path);
+      DoraMeta.FileStatus fs = mDoraWorker.buildFileStatusFromUfsStatus(status, path);
+      return Optional.ofNullable(fs);
+    } catch (FileNotFoundException e) {
+      return Optional.empty();
+    }
+  }
+
+  /**
+   * Gets file meta from UFS and loads it into metastore if exists.
+   * @param path the full ufs path
+   * @return the file status, or empty optional if not found
+   */
+  public Optional<FileStatus> loadFromUfs(String path) throws IOException {
+    Optional<FileStatus> fileStatus = getFromUfs(path);
+    fileStatus.ifPresent(status -> put(path, status));
+    return fileStatus;
+  }
+
+  /**
+   * Gets file meta from the metastore.
+   * @param path the full ufs path
+   * @return the file status, or empty optional if not found
+   */
+  public Optional<FileStatus> getFromMetaStore(String path) {
+    return mMetastore.getDoraMeta(path);
+  }
+
+  /**
+   * Puts meta of a file into the metastore, and invalidates the file data cache.
+   * @param path the full ufs path
+   * @param meta the file meta
+   */
+  public void put(String path, FileStatus meta) {
+    Optional<FileStatus> status = mMetastore.getDoraMeta(path);
+    if (status.isEmpty()
+        || status.get().getFileInfo().getFolder()
+        || status.get().getFileInfo().getLength() == 0) {
+      mMetastore.putDoraMeta(path, meta);
+      return;
+    }
+    if (mPopulateMetadataFingerprint) {
+      Fingerprint originalFingerprint =
+          Fingerprint.parse(status.get().getFileInfo().getUfsFingerprint());
+      if (originalFingerprint == null) {
+        originalFingerprint = Fingerprint.INVALID_FINGERPRINT;
+      }
+      Fingerprint newFingerprint =
+          Fingerprint.parse(meta.getFileInfo().getUfsFingerprint());
+      if (newFingerprint == null) {
+        newFingerprint = Fingerprint.INVALID_FINGERPRINT;
+      }
+      boolean contentMatched = originalFingerprint.isValid()
+          && newFingerprint.isValid()
+          && originalFingerprint.matchContent(newFingerprint);
+      if (!contentMatched) {
+        invalidateCachedFile(path, status.get().getFileInfo().getLength());
+      }
+    } else {
+      invalidateCachedFile(path, status.get().getFileInfo().getLength());
+    }
+    mMetastore.putDoraMeta(path, meta);
+  }
+
+  /**
+   * Removes meta of a file from the meta store.
+   * @param path the full ufs path
+   * @return the removed file meta, if exists
+   */
+  public Optional<FileStatus> removeFromMetaStore(String path) {
+    Optional<FileStatus> status = mMetastore.getDoraMeta(path);
+    if (status.isPresent()) {
+      mMetastore.removeDoraMeta(path);
+      invalidateCachedFile(path, status.get().getFileInfo().getLength());
+    }
+    return status;
+  }
+
+  private void invalidateCachedFile(String path, long length) {
```

Comment:
you can remove the now redundant method in `PagedDoraWorker` https://github.com/Alluxio/alluxio/blob/58617d0715500bd296898f11d03f10cf5ebd3bf0/dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java#L339-L352

Reply:
this is an interface method and i think it is still needed in the future. Though i changed the implementation a bit to make it cleaner